client/src/components/PostList/PostList.component.jsx

```diff
@@ -18,6 +19,54 @@ const PostList = ({ posts, defaultText, alertIfMoreThanDays, showViews }) => {
     { enabled: !!agencyShortName },
   )
   defaultText = defaultText ?? 'There are no posts to display.'
+
+  const googleAnalytics = useGoogleAnalytics()
+  const updateMaxScroll = () => {
+    googleAnalytics.maxScroll = Math.max(
+      googleAnalytics.maxScroll,
+      window.pageYOffset,
+    )
+  }
+
+  const sendMaxScrollToAnalytics = (maxScrollPossible) => {
+    maxScrollPossible = Math.max(maxScrollPossible, googleAnalytics.maxScroll) // this is required to restrict the percentage to 100
```

Comment:
If I'm not wrong, there are some browser compatibility issues which cause the total scroll height (i.e. `maxScrollPossible`) to differ on different browsers even with the same screen size (refer to [this post](https://stackoverflow.com/questions/17688595/finding-the-maximum-scroll-position-of-a-page)). There is a better API to track this ([window.scrollMaxY](https://developer.mozilla.org/en-US/docs/Web/API/Window/scrollMaxY)), but it's only available on Firefox so...

When the `maxScroll` exceeds `maxScrollPossible`, its not a lot usually within 20%. Since we're just using this as a gauge to see whether users browse, I think the precision doesn't have to be very high?

Reply:
I see! can we leave some documentation here (can link to the relevant SO post also) for future developers?